src/store/modules/user/actions.ts

```diff
@@ -204,9 +211,16 @@ const actions: ActionTree<UserState, RootState> = {
   /**
    * update current shopify config id
    */
-  async setCurrentShopifyConfig({ commit, dispatch }, shopifyConfig) {
-    commit(types.USER_CURRENT_SHOPIFY_CONFIG_UPDATED, shopifyConfig ? shopifyConfig : {});
-    dispatch('job/clearJobState', null, { root: true });
+  async setCurrentShopifyConfig({ commit, dispatch, state }, payload) {
+    if(payload.shopifyConfig) {
+      commit(types.USER_CURRENT_SHOPIFY_CONFIG_UPDATED, payload.shopifyConfig);
+    } else {
+      const shopifyConfigs = state.shopifyConfigs;
+      const currentShopifyConfig = shopifyConfigs.find((configs: any) => configs.shopifyConfigId === payload.shopifyConfigId)
+      commit(types.USER_CURRENT_SHOPIFY_CONFIG_UPDATED, currentShopifyConfig);
+    }
```

Comment:
I would suggest following code for better readability
```suggestion
    let currentShopifyConfig = payload.shopifyConfig;
    if(!currentShopifyConfig) {
      currentShopifyConfig = state.shopifyConfigs.find((shopifyConfig: any) => shopifyConfig.shopifyConfigId === payload.shopifyConfigId)
    } 
    commit(types.USER_CURRENT_SHOPIFY_CONFIG_UPDATED, currentShopifyConfig ? currentShopifyConfig : {});
```